proc_netstat.go

```diff
@@ -0,0 +1,440 @@
+	TCPDeferAcceptDrop        float64
+	IPReversePathFilter       float64
+	TCPTimeWaitOverflow       float64
+	TCPReqQFullDoCookies      float64
+	TCPReqQFullDrop           float64
+	TCPRetransFail            float64
+	TCPRcvCoalesce            float64
+	TCPOFOQueue               float64
+	TCPOFODrop                float64
+	TCPOFOMerge               float64
+	TCPChallengeACK           float64
+	TCPSYNChallenge           float64
+	TCPFastOpenActive         float64
+	TCPFastOpenActiveFail     float64
+	TCPFastOpenPassive        float64
+	TCPFastOpenPassiveFail    float64
+	TCPFastOpenListenOverflow float64
+	TCPFastOpenCookieReqd     float64
+	TCPFastOpenBlackhole      float64
+	TCPSpuriousRtxHostQueues  float64
+	BusyPollRxPackets         float64
+	TCPAutoCorking            float64
+	TCPFromZeroWindowAdv      float64
+	TCPToZeroWindowAdv        float64
+	TCPWantZeroWindowAdv      float64
+	TCPSynRetrans             float64
+	TCPOrigDataSent           float64
+	TCPHystartTrainDetect     float64
+	TCPHystartTrainCwnd       float64
+	TCPHystartDelayDetect     float64
+	TCPHystartDelayCwnd       float64
+	TCPACKSkippedSynRecv      float64
+	TCPACKSkippedPAWS         float64
+	TCPACKSkippedSeq          float64
+	TCPACKSkippedFinWait2     float64
+	TCPACKSkippedTimeWait     float64
+	TCPACKSkippedChallenge    float64
+	TCPWinProbe               float64
+	TCPKeepAlive              float64
+	TCPMTUPFail               float64
+	TCPMTUPSuccess            float64
+	TCPWqueueTooBig           float64
+}
+
+type IpExt struct {
+	InNoRoutes      float64
+	InTruncatedPkts float64
+	InMcastPkts     float64
+	OutMcastPkts    float64
+	InBcastPkts     float64
+	OutBcastPkts    float64
+	InOctets        float64
+	OutOctets       float64
+	InMcastOctets   float64
+	OutMcastOctets  float64
+	InBcastOctets   float64
+	OutBcastOctets  float64
+	InCsumErrors    float64
+	InNoECTPkts     float64
+	InECT1Pkts      float64
+	InECT0Pkts      float64
+	InCEPkts        float64
+	ReasmOverlaps   float64
+}
+
+func (p Proc) Netstat() (ProcNetstat, error) {
+	filename := p.path("net/netstat")
+	data, err := util.ReadFileNoStat(filename)
+	if err != nil {
+		return ProcNetstat{PID: p.PID}, err
+	}
+	procNetstat, err := parseNetstat(bytes.NewReader(data), filename)
+	procNetstat.PID = p.PID
+	return procNetstat, err
+}
+
+// parseNetstat parses the metrics from proc/<pid>/net/netstat file
+// and returns a ProcNetstat structure.
+func parseNetstat(r io.Reader, fileName string) (ProcNetstat, error) {
+	var (
+		scanner  = bufio.NewScanner(r)
+		procNetstat = ProcNetstat{}
+	)
+
+	for scanner.Scan() {
+		nameParts := strings.Split(scanner.Text(), " ")
+		scanner.Scan()
+		valueParts := strings.Split(scanner.Text(), " ")
+		// Remove trailing :.
+		protocol := nameParts[0][:len(nameParts[0])-1]
+		if len(nameParts) != len(valueParts) {
+			return procNetstat, fmt.Errorf("mismatch field count mismatch in %s: %s",
+				fileName, protocol)
+		}
+		for i := 1; i < len(nameParts); i++ {
```

Comment:
Could probably use a range over `nameParts` here instead of the explicit counter.

Reply:
Actually i need to pick elements from both `nameParts` and `valueParts`, i am not sure if a range over is possible here.

Reply:
hmm, I guess that wouldn't work because you want to start at index 1 instead of 0.  So you'd have to add something like
```
nameParts = nameParts[1:]
valueParts = valueParts[1:]
```
So I think just leaving it as with the explicit counter is fine, or whatever you prefer.

Reply:
@pgier I think yes, lets keep it with the explicit counter.